### ISSUE TITLE
How to use rtx with emacs?

### ISSUE THREAD

**Issue by adamcohen2** [jdxcode/rtx#638]
I'm wondering if there's a recommended way to use rtx with emacs? Right now I'm hardcoding the path to different binaries using the following, however, this is not ideal:

```lisp
(setenv "PATH" (concat "~/.local/share/rtx/installs/go/1.20.5/go/bin:" (getenv "PATH")))
'(flycheck-ruby-rubocop-executable "~/.local/share/rtx/installs/ruby/3.1.4/bin/rubocop")
```

Is there a better way to do this? Thanks!

**Comment by jimeh** [comment]
TL;DR: Add `~/.local/share/rtx/shims` to your `PATH` env var and `exec-path` variable within Emacs.

This will let Emacs find `~/.local/share/rtx/shims/rubocop` when looking for executable `rubocop`.

The shims directory contains executables for every binary across all installed tool-versions, and they run through rtx so it can determine what tool and version is relevant based on current path and any config files in place. The overhead rtx introduces is only around 10-15ms in my testing.

Personally I use the [exec-path-from-shell](https://github.com/purcell/exec-path-from-shell) package to setup paths and other env vars within Emacs based on my shell setup.

Here's the setup from [my config](https://github.com/jimeh/.emacs.d/blob/v0.8.45/core/siren-core-env.el#L10-L30):

```elisp
(use-package exec-path-from-shell
  :custom
  (exec-path-from-shell-variables '("PATH"
                                    "MANPATH"
                                    "TMPDIR"
                                    "KUBECONFIG"
                                    "GOPATH"
                                    "GOBIN"
                                    "GOROOT"
                                    "GOPRIVATE"
                                    "GOENV_GOPATH_PREFIX"
                                    "GOENV_VERSION"
                                    "RUST_BACKTRACE"
                                    "MallocNanoZone"))
  (exec-path-from-shell-arguments '("-l"))
  (exec-path-from-shell-check-startup-files nil)
  (exec-path-from-shell-debug nil)

  :config
  (when (memq window-system '(mac ns x pgtk))
    (exec-path-from-shell-initialize)))
```

Obviously add/remove env vars from the above list as you see fit. In fact, I should remove some myself that I no longer need :)

**Comment by adamcohen2** [comment]
> TL;DR: Add `~/.local/share/rtx/shims` to your `PATH` env var and `exec-path` variable within Emacs.

Awesome, thanks so much for the details @jimeh, this worked perfectly! It looks like it might be worth updating the [IDE Integration](https://github.com/jdxcode/rtx/#ide-integration) section in the readme to explain that using a package like [exec-path-from-shell](https://github.com/purcell/exec-path-from-shell) is also necessary for emacs.

**Comment by doolio** [comment]
> using a package like [exec-path-from-shell](https://github.com/purcell/exec-path-from-shell) is also necessary for emacs

It really depends on your OS. If your PATH is set as you want and your OS starts a login shell on login then `exec-path-ffrom-shell` is unnecessary. If you are using macOS or distributions based on Debian then it will be necessary.
